analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp6.cs

```diff
@@ -43,6 +43,20 @@ public void FlasePositive(object o)
                 var b = o.ToString(); // Compliant, o is checked for null in this branch
             }
             catch (ApplicationException) when (o == null)
+            {
+                var b = o.ToString(); // Unreachable, o? cannot throw ApplicationException
```

Comment:
But "ToString" could throw.

Reply:
`ToString` is only invoked if `o` is `NotNull`, and the `catch ... when (o == null)` makes it unreachable.

This is actually a pretty advanced case.